modules/ossm-deploying-automatic-gateway-injection.adoc

```diff
@@ -12,70 +12,191 @@ When deploying a gateway, you must opt-in to injection by adding an injection la
-  kind: RoleBinding
-  metadata:
-   name: istio-ingressgateway-sds
-   namespace: istio-ingress
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Enable gateway injection by setting the `sidecar.istio.io/inject` field to `"true"`. When connecting to a revisioned control plane, replace with `istio.io/rev: revision-name`.
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+selector:
+  matchLabels:
+    istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Set the `image` field to `auto` so that the image automatically updates each time the pod starts.
++
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-ingress
+spec:
+  selector:
+    matchLabels:
+      istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: <ingressgateway>
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: istio-proxy
+        image: auto
+----
+
+. Set up roles to allow reading credentials for TLS.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
   name: istio-ingressgateway-sds
+  namespace: istio-ingress
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+ name: istio-ingressgateway-sds
+ namespace: istio-ingress
+roleRef:
+apiGroup: rbac.authorization.k8s.io
+kind: Role
+name: istio-ingressgateway-sds
 subjects:
 - kind: ServiceAccount
-  name: default
+name: default
+----
+
+. Grant access to the new gateway from outside the cluster. This is required whenever `spec.security.manageNetworkPolicy` is set to `true`.
++
+[source,yaml]
+----
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: gatewayingress
+  namespace: istio-ingress
+spec:
+  podSelector:
+    matchLabels:
+      istio: ingressgateway
+  ingress:
+    - {}
+  policyTypes:
+  - Ingress
+----
+
+. Automatically scale the pod when ingress traffic increases. This example sets the minimum replicas to 2, the maximum replicas to 5, and creates another replica when utilization reaches 80%.
++
+[source,yaml]
+----
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  labels:
+    istio: <ingressgateway>
```

Comment:
There's a few references to '<ingressgateway>' I think we should either give a full example using an actual name and stick to it e.g. 'istio-ingressgateway' or have placeholders, but if we have placeholders can we just say 'gateway' as the gateways could be for ingress or egress.